### PR TITLE
[WIP] user and admin roles, initial user creation, permission example

### PR DESCRIPTION
**Pull request by henderkes**
References issues #112 #110 #109

### CODE REVIEW

**Review by Drakulix — COMMENTED** [review]
Very good start. I am not sure however if we should merge this in the current state, when there is no way to login, the initial user creation is this simple and this seems to depend on code of another PR of yours. Also there are pyc-files.

But really solid, I like the way this is going!

**Comment by henderkes** [comment]
yeah, I accidentally created this branch from within the project-post-adjustment branch which resulted in the first 8 commits, it's however not reliant on the other PR, I can do a rebase from master if needed

**Comment by henderkes** [comment]
rewrote history, should be conform with master now and not use the changes of #125

**Comment by henderkes** [comment]
for some reason it still lets me access @login_required protected pages without logging in.
I will look into that and build a basic login page as soon as I find time

also references issue #130 now. collaborators are welcome, feel free to assign yourself

File: backend/flask_api.py [hunk 1]
```diff
@@ -30,6 +29,74 @@
 ALLOWED_EXTENSIONS = {'txt', 'json', 'json5'}
 app.config['UPLOAD_FOLDER'] = ''
 app.config['MAX_CONTENT_PATH'] = 1000000  # 100.000 byte = 100kb
+app.config['MONGODB_DB'] = 'knexDB'
+app.config['MONGODB_HOST'] = 'mongodb'
+app.config['MONGODB_PORT'] = 27017
+
+es = Elasticsearch([{'host': 'elasticsearch', 'port': 9200}])
+
+client = MongoClient('mongodb:27017')
```
Comment (markwitz) [inline backend/flask_api.py:38]: `client = MongoClient('mongodb:27017')` should'nt the client get the connection by using the config?
```
+app.config['MONGODB_DB'] = 'knexDB'
+app.config['MONGODB_HOST'] = 'mongodb'
+app.config['MONGODB_PORT'] = 27017
```
if u are startig to use the config at this point we should use it as well

Comment (henderkes) [reply backend/flask_api.py:38]: had to do that for the MongoEngine, but yes I agree, a good idea to use it for the project db client as well

would look like an abomination though, I guess there's no direct constructor

**Pull request closed without merging**